packages/firestore/test/integration/remote/stream.test.ts

```diff
@@ -262,4 +281,51 @@ describe('Write Stream', () => {
         });
     }, queue);
   });
+
+  it('force refreshes auth token on receiving unauthenticated error', () => {
+    const queue = new AsyncQueue();
+    const credentials = new MockCredentialsProvider();
+
+    return withTestDatastore(
+      ds => {
+        const writeStream = ds.newPersistentWriteStream();
+        writeStream.start(streamListener);
+        return streamListener
+          .awaitCallback('open')
+          .then(() => {
+            // Simulate callback from GRPC with an unauthenticated error -- this should invalidate
+            // the token.
+            writeStream.handleStreamClose(
+              new FirestoreError(Code.UNAUTHENTICATED, '')
+            );
+            return streamListener.awaitCallback('close');
+          })
+          .then(() => {
+            writeStream.start(streamListener);
+            return streamListener.awaitCallback('open');
+          })
+          .then(() => {
+            // Simulate a different error -- token should not be invalidated this time.
+            writeStream.handleStreamClose(
+              new FirestoreError(Code.UNAVAILABLE, '')
```

Comment:
This might be a bad idea -- feel free to push back. The compiler complains that `handleStreamClose` is private, but `yarn test` can run this test just fine. It's clear that the build process ignores the compile-time error, but I don't know if it's possible to rely on that.